README.md

```diff
@@ -17,15 +17,80 @@ The latest stable releases on master support the pancancer project https://icgc.
 
 See the Consonance [wiki](https://github.com/Consonance/consonance/wiki) for more information on this project.
 
-## Building
+## Dependencies
+
+I'm showing how to install dependencies on a Mac so adapt to your system/OS.
+
+### PostgreSQL
+
+You need a PostgreSQL server to run the tests, on the Mac try [Postgres.app](http://postgresapp.com/).
+
+Once you have this setup (using whatever technique is appropriate for your system) create the `queue_status` database.
+
+    $bash> "/Applications/Postgres.app/Contents/Versions/10/bin/psql" -p5432 -d "postgres"
+    postgres=# create database queue_status;
+    postgres=# create user queue_user;
+    postgres=# grant all privileges  on database queue_status to queue_user;
+
+Now load the schema:
+
+    $bash> cat ~/gitroot/Consonance/consonance/consonance-arch/sql/schema.sql | "/Applications/Postgres.app/Contents/Versions/10/bin/psql" -p5432 -d "queue_status"
+
+### RabbitMQ
+
+The integration tests require RabbitMQ, here's how to install
+via home brew:
+
+    $bash> brew install rabbitmq
+    $bash> /usr/local/sbin/rabbitmq-server
+
+### Consonance Config File
+
+By default the client looks for `~/.consonance/config`. You need this file for
+tests to pass:
+
+    $bash> mkdir ~/.consonance/ && cp ./consonance-client/src/test/resources/config ~/.consonance/config
+
+### Dockstore CLI
+
+The integration tests (see below for how to trigger) will actually simulate the full lifecycle of a WDL workflow run using the Dockstore CLI.  See http://dockstore.org for how to install, you need the `dockstore` command dependencies in your path and properly configured in order for the full integration tests to pass.  Consonance is using the Dockstore CLI library so you only need to install cwltool in your path if you want to test CWL workflows.  Cromwell for WDL workflows is baked into the Dockstore CLI library.
+
+### Docker
+
+Related to the Dockstore CLI above, one of the WDL workflows used for testing requires Docker.  Make sure the system you are using for testing has Docker installed if you run the full integration tests.
+
+## Building & Testing
 
 The build uses maven (3.2.3, look at using [MVNVM](http://mvnvm.org/)), just run:
 
-    mvn clean install
+    $bash> mvn clean install
 
  To avoid tests (probably a bad idea!):
 
-    mvn -DskipTests=true clean install
+    $bash> mvn -Dmaven.test.skip=true clean install
+
+Skip tests and use parallel build (see more info [here](https://zeroturnaround.com/rebellabs/your-maven-build-is-slow-speed-it-up/)):
+
+    $bash> mvn -Dmaven.test.skip=true -T 1C install -pl consonance-integration-testing -am
```

Comment:
If I'm reading this right. This might skip compilation of a bunch of projects which may be confusing if not documented 
```
 -pl,--projects <arg>                   Comma-delimited list of specified
                                        reactor projects to build instead
                                        of all projects.
```